test/testParachain.t.sol

```diff
@@ -46,25 +54,251 @@ contract ParachainTest is Test {
         vm.etch(_address, deployed.code);
     }
 
-    function testConstructor() public {}
+    function testConstructor() public {
+        assertEq(address(registry), parachain.registryAddress());
+    }
+
+    function testReportStakeDeposited() public {
+        // setup
+        IRegistry.Parachain memory fakeParachain = IRegistry.Parachain({
+            id: fakeParaId,
+            owner: paraOwner,
+            palletInstance: abi.encode(fakePalletInstance)
+        });
+        IRegistry.Parachain memory badFakeParachain = IRegistry.Parachain({
+            id: fakeParaId,
+            owner: address(0),
+            palletInstance: abi.encode(fakePalletInstance)
+        });
+        uint256 fakeAmount = 100e18;
+
+        // test non-registered parachain - should revert with "Parachain not registered"
+        vm.expectRevert();
```

Comment:
instead of having the revert string in the comment, I think you can just do `vm.expectRevert("Parachain not registered");`